src/Amplitude/Event.php

```diff
@@ -4,29 +4,58 @@
 
 use RuntimeException;
 
+/**
+ * @phpstan-type Payload array{
+ *     event_type: string,
+ *     event_properties?: array<mixed>|null,
+ *     user_properties?: array<mixed>|null,
+ *     user_id?: string|null,
+ *     device_id?: string|null,
+ *     insert_id?: string|null,
+ *     time?: int|null
+ * }
+ */
 class Event
 {
-    public ?string $eventType = null;
+    public ?string $eventType;
     /**
-     * @var ?array<mixed>
+     * @var array<mixed>|null
      */
-    public ?array $eventProperties = null;
+    public ?array $eventProperties;
     /**
-     * @var ?array<mixed>
+     * @var array<mixed>|null
      */
-    public ?array $userProperties = null;
-    public ?string $userId = null;
-    public ?string $deviceId = null;
-    public ?string $insertId = null;
-    public ?int $time = null;
+    public ?array $userProperties;
+    public ?string $userId;
+    public ?string $deviceId;
+    public ?string $insertId;
+    public ?int $time;
 
-    public function __construct(string $eventType)
+    /**
+     * @param array<mixed>|null $eventProperties
+     * @param array<mixed>|null $userProperties
+     */
+    public function __construct(
+        string $eventType,
+        ?array $eventProperties = null,
+        ?array $userProperties = null,
+        ?string $userId = null,
+        ?string $deviceId = null,
+        ?string $insertId = null,
+        ?int $time = null
+    )
     {
         $this->eventType = $eventType;
+        $this->eventProperties = $eventProperties;
+        $this->userProperties = $userProperties;
+        $this->userId = $userId;
+        $this->deviceId = $deviceId;
+        $this->insertId = $insertId;
+        $this->time = $time;
     }
 
     /**
-     * @return array<string, mixed>
+     * @return Payload
      */
     public function toArray(): array
```

Comment:
Suggested update:

```
public function toArray(): array
    {
        return array_filter([
            'event_type' => $this->eventType,
            'event_properties' => $this->eventProperties,
            'user_properties' => $this->userProperties,
            'user_id' => $this->userId,
            'device_id' => $this->deviceId,
            'insert_id' => $this->insertId,
            'time' => $this->time,
        ], fn($value) => $value !== null);
    }
```

Reply:
When you don't pass a callable to array_filter, it does filter anything that is null. So there is no point in providing this callable, as it does the same, but only being explicit.